### PR TITLE
Clean up OWNERS for pkg/probe

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Adds sig/node labels to PRs in pkg/probe.

Updates approvers to match the SIG alias.

/sig node
/cc @derekwaynecarr @dchen1107 

#### Which issue(s) this PR fixes:

N/A, cleanup.

/priority backlog

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/probe/OWNERS
Comment (SergeyKanzhelev) [inline pkg/probe/OWNERS:4]: this removes @tallclair from the list. Intentional?

Comment (pacoxu) [reply pkg/probe/OWNERS:4]: https://github.com/kubernetes/kubernetes/commit/933a2ca634cc71b77e907558df68920f7c143629

```
 # see https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES#LC220:~:text=sig%2Dnode%2Dreviewers
# see https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES#LC220:~:text=sig%2Dnode%2Dapprovers
```

The comments help when I read it the first time.

**Review by troy0820 — APPROVED** [review]

File: pkg/probe/OWNERS
Comment (SergeyKanzhelev) [reply pkg/probe/OWNERS:4]: /hold

holding to make sure that removing one approver is intentional. Adding more like @mrunalp who cannot approve this PR is definitely an improvement. Feel free to un-hold

Comment (ehashman) [reply pkg/probe/OWNERS:4]: I'll quickly ask @tallclair, I don't think he's been active in this code but he's the only effective removal.

Comment (ehashman) [reply pkg/probe/OWNERS:4]: /hold cancel

He said he's fine being removed :)

**Pull request merged**
